### PR TITLE
ENG-7370: @AdHoc doesn't choke on -- comments.

### PR DESCRIPTION
**Pull request by mikeismert**

### CODE REVIEW

File: src/frontend/org/voltdb/utils/SQLLexer.java [hunk 1]
```diff
@@ -25,6 +25,14 @@
 public class SQLLexer
 {
     private static final VoltLogger COMPILER_LOG = new VoltLogger("COMPILER");
+
+    // Match single-line comments
+    private static final Pattern SINGLE_LINE_COMMENT_MATCH = Pattern.compile(
+            "^\\s*" + // start of line, 0 or more whitespace
+            "--" + // start of comment
+            ".*$", + // everything to end of line
+            Pattern.CASE_INSENSITIVE);
```
Comment (wijjo) [inline src/frontend/org/voltdb/utils/SQLLexer.java]: nit: don't need CASE_INSENSITIVE

File: src/frontend/org/voltdb/compiler/AsyncCompilerAgent.java [hunk 2]
```diff
@@ -121,6 +121,9 @@ void handleMailboxMessage(final VoltMessage message) {
             SortedSet<String> conflictTables = new TreeSet<String>();
             Set<String> createdTables = new HashSet<String>();
             for (String stmt : w.sqlStatements) {
+                if (SQLLexer.isComment(stmt)) {
```
Comment (wijjo) [inline src/frontend/org/voltdb/compiler/AsyncCompilerAgent.java]: I'm probably not putting it all together in my head... But if the splitter is pre-filtering comments why is this check needed?

File: src/frontend/org/voltdb/utils/MiscUtils.java [hunk 3]
```diff
@@ -646,9 +646,25 @@ public static synchronized void printPortsInUse(VoltLogger log) {
...
         // IMPORTANT: The loop is structured in a way that requires all if/else/... blocks to bump
         // iCur appropriately. Failure of a corner case to bump iCur will cause an infinite loop.
+        boolean statementIsComment = false;
+        boolean inStatement = false;
         int iCur = 0;
         while (iCur < buf.length) {
-            if (sCommentEnd != null) {
+            // Eat up whitespace outside of a statement
+            if (!inStatement) {
+                if (buf[iCur] == ' ' ||
```
Comment (wijjo) [inline src/frontend/org/voltdb/utils/MiscUtils.java]: Would it be better to use Character.isSpaceChar(ch)?

Comment (mikeismert) [inline src/frontend/org/voltdb/utils/MiscUtils.java]: Sure?  Looking, any reason not to go wholesale for isWhitespace(ch)?

Comment (wijjo) [inline src/frontend/org/voltdb/utils/MiscUtils.java]: That was just the one Google found for me. 



On Thu, Jan 8, 2015 at 11:56 AM -0800, "Mike Ismert" <notifications@github.com> wrote:












In src/frontend/org/voltdb/utils/MiscUtils.java:
>          int iCur = 0;
>          while (iCur < buf.length) {
> -            if (sCommentEnd != null) {
> +            // Eat up whitespace outside of a statement
> +            if (!inStatement) {
> +                if (buf[iCur] == ' ' ||



Sure?  Looking, any reason not to go wholesale for isWhitespace(ch)?

**Comment by jhugg** [comment]
I can see the meh, but it seems like it should be fine? I like Steve's comment about Character.isWhitespace or whatever. 

Sadly this is going to conflict with my ddl-diff-smahts pull.

**Comment by wijjo** [comment]
Looks fine. You can address the 2 nits or not. Up to you.

**Comment by mikeismert** [comment]
Okay, I'll clean them up and do a squash merge.  Closing the pull request.

**Pull request closed without merging**